frameworks/react/src/AdvancedVideo.tsx

```diff
@@ -0,0 +1,126 @@
+} from '@cloudinary/html';
+
+type ReactEventHandler<T = Element> = EventHandler<SyntheticEvent<T>>;
+
+interface VideoProps {
+  cldvid: CloudinaryVideo,
+  plugins?: Plugins,
+  sources?: VideoSources,
+
+  // supported video attributes
+  controls?: boolean
+  loop?: boolean,
+  muted?: boolean,
+  poster?: string,
+  preload?: string,
+  autoPlay?: boolean,
+  playsInline?: boolean
+
+  // supported video events
+  onPlay?: ReactEventHandler<any>,
+  onLoadStart?: ReactEventHandler<any>,
+  onPlaying?: ReactEventHandler<any>,
+  onError?: ReactEventHandler<any>,
+  onEnded?: ReactEventHandler<any>
+}
+
+/**
+ * @memberOf ReactSDK
+ * @type {Component}
+ * @description The Cloudinary video component
+ * @prop {CloudinaryVideo} transformation Generated by @cloudinary/base
+ * @prop {Plugins} plugins Advanced image component plugins lazyload()
+ * @prop videoAttributes Optional attributes include controls, loop, muted, poster, preload, autoplay
+ * @prop videoEvents Optional video events include play, loadstart, playing, error, ended
+ * @prop {VideoSources} sources Optional sources to generate
+ * @example
+ *  <caption>
+ *  Using custom defined resources.
+ * </caption>
+ *   vid = new CloudinaryVideo('dog', {cloudName: 'demo'});
+ *   sources = [
+ {
+        type: 'mp4',
+        codecs: ['vp8', 'vorbis'],
+        videoCodec: 'auto'
+      },
+ {
+        type: 'webm',
+        codecs: ['avc1.4D401E', 'mp4a.40.2'],
+        videoCodec: 'auto'
+      }];
+ *
+ * <AdvancedVideo cldvid={vid} controls sources={sources}/>
+ */
+class AdvancedVideo extends React.Component <VideoProps> {
+  videoRef: React.RefObject<HTMLVideoElement>;
+  htmlVideoLayerInstance: HtmlVideoLayer;
+  private videoAttributes: object;
+
+  constructor(props: VideoProps) {
+    super(props);
+    this.videoRef = React.createRef();
+  }
+
+  /**
+   * On mount creates a new HTMLVideoLayer instance and initialises with ref to video element,
+   * user generated cloudinaryVideo and the plugins to be used
+   */
+  componentDidMount() {
+    this.videoAttributes = {
+      controls: this.props.controls,
+      loop: this.props.loop,
+      muted: this.props.muted,
+      poster: this.props.poster,
+      preload: this.props.preload,
+      autoplay: this.props.autoPlay,
+      playsinline: this.props.playsInline
+    };
+
+    this.htmlVideoLayerInstance = new HtmlVideoLayer(
+      this.videoRef.current,
+      this.props.cldvid,
+      this.props.sources,
+      this.props.plugins,
+      this.videoAttributes
+    )
+  }
+
+  /**
+   * On update we cancel running plugins and update the video instance if the src
+   * was changed
+   */
+  componentDidUpdate() {
+    cancelCurrentlyRunningPlugins(this.htmlVideoLayerInstance.htmlPluginState);
+    // call html layer to update the dom again with plugins and reset toBeCanceled
+    this.htmlVideoLayerInstance.update(this.props.cldvid, this.props.sources, this.props.plugins, this.videoAttributes)
```

Comment:
video attributes might have changed no?

Reply:
you're right! updated to an outside method.. lmk if you think there is a better way to do this